examples/finding_sunspots_using_stara.py

```diff
@@ -0,0 +1,130 @@
+
+cont_rotated = hmi_map.rotate(order=3)
+
+###############################################################################
+# Plot the rotated map.
+
+fig = plt.figure()
+ax = plt.subplot(projection=cont_rotated)
+im = cont_rotated.plot(axes=ax)
+
+###############################################################################
+# To reduce computational expense, we resample the continuum image to a lower
+# resolution. For computational perspective, the ``hmi_map`` has a dimension of ``4102*4102`` pixels.
+# The next step downsamples this continuum image to a dimension of ``1024*1024`` pixels,
+# reducing the data size by nearly 16 times. This step ensures that running the algorithm on the full-resolution
+# image is not overly computationally expensive.
+
+cont_rotated_resample = cont_rotated.resample((1024, 1024) * u.pixel)
+
+###############################################################################
+# Next, we use the STARA function to detect sunspots in the resampled map.
+
+segs = stara(cont_rotated_resample, limb_filter=10 * u.percent)
+
+###############################################################################
+# Finally, we plot the resampled map along with the detected sunspots. We create
+# a new Matplotlib figure and subplot with the projection defined by the resampled
+# map, plot the resampled map, and overlay contours of the detected sunspots using
+# the ``contour`` method.
+
+fig = plt.figure()
+ax = plt.subplot(projection=cont_rotated_resample)
+im = cont_rotated_resample.plot(axes=ax, autoalign=True)
+ax.contour(segs, levels=0)
+
+###############################################################################
+# To focus on specific regions containing sunspots, we can create a submap,
+# which is a smaller section of the original map. This allows us to zoom in
+# on areas of interest. We define the coordinates of the rectangle to crop
+# in pixel coordinates.
+
+bottom_left = cont_rotated_resample.pixel_to_world(240 * u.pix, 350 * u.pix)
+top_right = cont_rotated_resample.pixel_to_world(310 * u.pix, 410 * u.pix)
+
+# Create the submap using the world coordinates of the bottom left and top right corners.
+hmi_submap = cont_rotated_resample.submap(bottom_left, top_right=top_right)
+segs = stara(hmi_submap, limb_filter=10 * u.percent)
+
+# Plot the submap along with the contours.
+fig = plt.figure()
+ax = plt.subplot(projection=hmi_submap)
+im = hmi_submap.plot(axes=ax, autoalign=True)
+ax.contour(segs, levels=0)
+
+plt.show()
+
+###############################################################################
+# We can further enhance our analysis by extracting key properties from the
+# segmented image and organizing them into a structured table.
+
+# First, a labeled image is created where each connected component (sunspot)
+# is assigned a unique label.
+labelled = label(segs)
+# If no sunspots are detected (labelled.max() == 0), print an empty table.
+if labelled.max() == 0:
+    print(QTable())
+else:
```

Comment:
Do we need this if else block?

Reply:
for this example it isn't necessary. I'll remove this.